diff -ruN ncl-6.6.2/ni/src/ncl/NclApi.c ncl-6.6.2-patched/ni/src/ncl/NclApi.c
--- ncl-6.6.2/ni/src/ncl/NclApi.c	2019-02-28 00:44:39.000000000 +0100
+++ ncl-6.6.2-patched/ni/src/ncl/NclApi.c	2021-09-23 19:36:33.146606149 +0200
@@ -140,9 +140,9 @@
 extern char *the_input_buffer_ptr;
 extern int the_input_buffer_size;
 
-FILE *thefptr;
-FILE *theoptr;
-int cmd_line;
+extern FILE *thefptr;
+extern FILE *theoptr;
+extern int cmd_line;
 extern int cur_line_number;
 extern char *cur_line_text;
 extern int cur_line_maxsize;
diff -ruN ncl-6.6.2/ni/src/ncl/NclNewHDF5.c ncl-6.6.2-patched/ni/src/ncl/NclNewHDF5.c
--- ncl-6.6.2/ni/src/ncl/NclNewHDF5.c	2019-02-28 00:44:39.000000000 +0100
+++ ncl-6.6.2-patched/ni/src/ncl/NclNewHDF5.c	2021-09-23 19:41:03.919542763 +0200
@@ -82,7 +82,7 @@
 
 #define NUMPOSDIMNAMES	6
 
-NclQuark possibleDimNames[NUMPOSDIMNAMES];
+extern NclQuark possibleDimNames[NUMPOSDIMNAMES];
 
 #ifndef FALSE
 #define FALSE           0
