diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS -x autom4te.cache ncl-6.6.2.pre-int-conversion/ni/src/lib/hlu/TextItem.c ncl-6.6.2/ni/src/lib/hlu/TextItem.c
--- ncl-6.6.2.pre-int-conversion/ni/src/lib/hlu/TextItem.c	2023-05-23 17:53:44.629179029 +0200
+++ ncl-6.6.2/ni/src/lib/hlu/TextItem.c	2023-05-23 17:59:59.966837608 +0200
@@ -63,7 +63,7 @@
 		NhlTProcedure,_NhlUSET((NhlPointer)_NhlResUnset),0,NULL },
 	{ NhlNtxFont, NhlCFont, NhlTFont, sizeof(NhlFont),
 		NhlOffset(NhlTextItemLayerRec, text.font),
-		NhlTImmediate,_NhlUSET(21),0,NULL },
+		NhlTImmediate,_NhlUSET((NhlPointer)21),0,NULL },
 	{ "no.res", "No.res", NhlTBoolean, sizeof(NhlBoolean),
 		NhlOffset(NhlTextItemLayerRec,text.just_set),
 		NhlTImmediate,_NhlUSET((NhlPointer)True),_NhlRES_PRIVATE,NULL},
diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS -x autom4te.cache ncl-6.6.2.pre-int-conversion/ni/src/ncl/NclNetCdf.c ncl-6.6.2/ni/src/ncl/NclNetCdf.c
--- ncl-6.6.2.pre-int-conversion/ni/src/ncl/NclNetCdf.c	2023-05-23 17:53:44.790179741 +0200
+++ ncl-6.6.2/ni/src/ncl/NclNetCdf.c	2023-05-23 17:53:48.855197705 +0200
@@ -251,7 +251,7 @@
 			*(nc_type*)out_type = NC_LONG;
 		} else {
 			NhlPError(NhlWARNING,NhlEUNKNOWN,"Classic NetCDF does not support 8-byte longs; convert to a supported data type or use the NetCDF4 format");
-			*(nc_type*)out_type = NULL;
+			*(nc_type*)out_type = 0;
 		}
 		break;
 	case NCL_float:
