diff -ruN ncl-6.6.2.orig/ni/src/nyacc/yMakefile ncl-6.6.2/ni/src/nyacc/yMakefile
--- ncl-6.6.2.orig/ni/src/nyacc/yMakefile	2019-02-27 17:44:39.000000000 -0600
+++ ncl-6.6.2/ni/src/nyacc/yMakefile	2024-04-21 14:13:02.179370458 -0600
@@ -1,3 +1,7 @@
+#if defined(Darwin)
+ EXTRA_CCOPTIONS = -std=gnu89
+#endif
+
 MYBIN		= nyacc
 
 SRCS		= closure.c lalr.c main.c output.c skeleton.c verbose.c \
diff -ruN ncl-6.6.2.orig/yMakefile ncl-6.6.2/yMakefile
--- ncl-6.6.2.orig/yMakefile	2019-02-27 17:44:39.000000000 -0600
+++ ncl-6.6.2/yMakefile	2024-04-22 09:12:11.731692388 -0600
@@ -29,6 +29,10 @@
 SUBDIRS	= config common ncarg2d ngmath ncarview
 #endif
 
+#if defined(Darwin)
+ EXTRA_CCOPTIONS = -std=gnu89 -Wno-error=int-conversion
+#endif
+
 #ifdef	YMAKEDEVFILE
 Info::
 	@echo;\
