diff -ruN -x __pycache__ -x obj -x '*~' -x tags -x TAGS ncl-6.6.2.orig/config/Makefile.ini ncl-6.6.2/config/Makefile.ini
--- ncl-6.6.2.orig/config/Makefile.ini	2019-02-28 00:44:39.000000000 +0100
+++ ncl-6.6.2/config/Makefile.ini	2023-02-28 15:24:49.955887147 +0100
@@ -24,7 +24,7 @@
 all: ymake-filter
 
 ymake-filter: ymake-filter.o
-	$(CC) $(CFLAGS) -o ymake-filter ymake-filter.o
+	$(CC_FOR_BUILD) $(CFLAGS) -o ymake-filter ymake-filter.o
 
 clean:
 	@-rm -f ymake-filter ymake-filter.o core
